nil pointer: FrameManager.NavigateFrame

### Brief summary

Nil pointer exception occurred in [FrameManager.NavigateFrame](https://github.com/grafana/xk6-browser/blob/d68946077680ff60f8b741f9305ce6dcde3fd78b/common/frame_manager.go#L588).

Test Run ID: 2094567

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x11e1f32]

goroutine 10318 [running]:
github.com/grafana/xk6-browser/common.(*FrameManager).NavigateFrame.func1({0x1489ce0?, 0xc001879880})
	github.com/grafana/xk6-browser@v1.1.1-0.20231026144352-d68946077680/common/frame_manager.go:588 +0x52
github.com/grafana/xk6-browser/common.createWaitForEventHandler.func1()
	github.com/grafana/xk6-browser@v1.1.1-0.20231026144352-d68946077680/common/helpers.go:145 +0x1e4
created by github.com/grafana/xk6-browser/common.createWaitForEventHandler in goroutine 10317
	github.com/grafana/xk6-browser@v1.1.1-0.20231026144352-d68946077680/common/helpers.go:137 +0x165
```

### xk6-browser version

v1.1.1

### OS

remote

### Chrome version

remote

### Docker version and image (if applicable)

_No response_

### Steps to reproduce the problem

Run test locally.

### Expected behaviour

No panics.

### Actual behaviour

Panics.

--- a/common/frame_manager.go
+++ b/common/frame_manager.go
@@ -584,7 +584,9 @@ func (m *FrameManager) NavigateFrame(frame *Frame, url string, parsedOpts *Frame
 		func(data any) bool {
 			newDocID := <-newDocIDCh
 			if evt, ok := data.(*NavigationEvent); ok {
-				return evt.newDocument.documentID == newDocID
+				if evt.newDocument != nil {
+					return evt.newDocument.documentID == newDocID
+				}
 			}
 			return false
 		})

